addons/api/addon/models/credential.js

```diff
@@ -3,7 +3,7 @@ import GeneratedCredentialModel from '../generated/models/credential';
 /**
  * Supported Credential types.
  */
-export const types = ['username_password', 'ssh_private_key'];
+export const types = ['username_password', 'ssh_private_key', 'json'];
```

Comment:
We can leave JSON credentials unknown until the we have the forms created.
```suggestion
export const types = ['username_password', 'ssh_private_key'];
```